src/DynamoCore/Logging/DynamoAnalyticsClient.cs

```diff
@@ -174,11 +180,16 @@ public DynamoAnalyticsClient(DynamoModel dynamoModel)
             //reporting approved status.
             preferences = dynamoModel.PreferenceSettings;
 
+            adpAnalyticsUI = new ADPAnalyticsUI();
+
             if (Session == null) Session = new DynamoAnalyticsSession();
 
             //Setup Analytics service, StabilityCookie, Heartbeat and UsageLog.
             Session.Start(dynamoModel);
 
+            Service.Instance.AddTrackerFactoryFilter(GATrackerFactory.Name, () => true == ReportingAnalytics);
+            Service.Instance.AddTrackerFactoryFilter(ADPTrackerFactory.Name, () => true == ReportingADPAnalytics);
+
```

Comment:
These filters will ensure that if ADP is **off** then no events will be passed to ADP. Same for google, if GoogleAnalytics are turned off...then no events will be processed by the GoogleTracker.

Reply:
Thanks. That mechanism implementation is exellent. There are some other aspects, let's talk offline